pkgs/applications/window-managers/jwm/default.nix

```diff
@@ -1,22 +1,33 @@
-{ stdenv, fetchurl, libX11, libXext, libXinerama, libXpm, libXft, freetype,
+{ stdenv, fetchurl, pkgconfig, automake, autoconf, libtool, libX11, libXext,
+  libXinerama, libXpm, libXft, which, xorg, libXau, libXdmcp, libpng, libjpeg,
+  expat, xproto, xextproto, xineramaproto, librsvg, gettext, freetype,
   fontconfig }:
 
 stdenv.mkDerivation rec {
-  name = "jwm-2.2.2";
+  name = "jwm-${version}";
+  version = "s1406";
   
   src = fetchurl {
-     url = "http://www.joewing.net/projects/jwm/releases/${name}.tar.xz";
-     sha256 = "0nhyy78c6imk85d47bakk460x0cfhkyghqq82zghmb00dhwiryln";
+     url = "https://github.com/joewing/jwm/archive/${version}.tar.gz";
+     sha256 = "0yk22b7cshhyfpcqnb4p59yxspx95xg9yp1kmkxi2fyw95cacab4";
   };
 
-  buildInputs = [ libX11 libXext libXinerama libXpm libXft freetype 
-    fontconfig ];
+  buildInputs = [
+    pkgconfig automake autoconf libtool libX11 libXext libXinerama libXpm
+    libXft which xorg.libXrender libXau libXdmcp libpng libjpeg expat xproto
+    xextproto xineramaproto librsvg gettext freetype fontconfig
+  ];
```

Comment:
Some of these are `nativeBuildInputs` (pkgconfig, auto*). `buildInputs` are dependencies needed by the actual program during runtime while `nativeBuildInputs` are needed to build the derivation. For example a x86 linux machine needs native `gcc` to build for arm.